pkg/cache/snapshot_test.go

```diff
@@ -245,148 +293,157 @@ func TestSnapshot(t *testing.T) {
+								corev1.ResourceCPU: {
+									{
+										Name: "default",
+										Min:  100_000,
+									},
+								},
+							},
+							UsedResources: Resources{
+								corev1.ResourceCPU: {"default": 0},
+							},
+							Workloads: map[string]*workload.Info{},
+						},
 					},
-				},
-				Workloads: map[string]*workload.Info{
-					"/beta":  workload.NewInfo(&workloads[1]),
-					"/gamma": workload.NewInfo(&workloads[2]),
-				},
-				NamespaceSelector: labels.Nothing(),
-				LabelKeys:         map[corev1.ResourceName]sets.String{corev1.ResourceCPU: {"baz": {}, "instance": {}}},
-				Status:            Active,
-			},
-			"bar": {
-				Name: "bar",
-				RequestableResources: map[corev1.ResourceName][]FlavorLimits{
-					corev1.ResourceCPU: {
-						{
-							Name: "default",
-							Min:  100_000,
+					ResourceFlavors: map[string]*kueue.ResourceFlavor{
+						"demand": {
+							ObjectMeta: metav1.ObjectMeta{Name: "demand"},
+							Labels:     map[string]string{"foo": "bar", "instance": "demand"},
+						},
+						"spot": {
+							ObjectMeta: metav1.ObjectMeta{Name: "spot"},
+							Labels:     map[string]string{"baz": "bar", "instance": "spot"},
+						},
+						"default": {
+							ObjectMeta: metav1.ObjectMeta{Name: "default"},
 						},
 					},
-				},
-				UsedResources: Resources{
-					corev1.ResourceCPU: map[string]int64{"default": 0},
-				},
-				Workloads:         map[string]*workload.Info{},
-				NamespaceSelector: labels.Nothing(),
-				Status:            Active,
+				}
 			},
 		},
-		ResourceFlavors: map[string]*kueue.ResourceFlavor{
-			"default": {
-				ObjectMeta: metav1.ObjectMeta{Name: "default"},
-				Labels:     nil,
-			},
-			"demand": {
-				ObjectMeta: metav1.ObjectMeta{Name: "demand"},
-				Labels:     map[string]string{"foo": "bar", "instance": "demand"},
-			},
-			"spot": {
-				ObjectMeta: metav1.ObjectMeta{Name: "spot"},
-				Labels:     map[string]string{"baz": "bar", "instance": "spot"},
-			},
-		},
-		InactiveClusterQueueSets: sets.String{"flavor-nonexistent-cq": {}},
 	}
-	if diff := cmp.Diff(wantSnapshot, snapshot, cmpopts.IgnoreUnexported(Cohort{})); diff != "" {
-		t.Errorf("Unexpected Snapshot (-want,+got):\n%s", diff)
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			cache := New(fake.NewClientBuilder().WithScheme(scheme).Build())
+
+			for _, c := range tc.clusterQueues {
+				// Purposely do not make a copy of clusterQueues. Clones of necessary fields are
+				// done in AddClusterQueue.
+				if err := cache.AddClusterQueue(context.Background(), c); err != nil {
+					t.Fatalf("Failed adding ClusterQueue: %v", err)
+				}
+			}
+
+			for i := range tc.flavors {
+				cache.AddOrUpdateResourceFlavor(&tc.flavors[i])
+			}
+
+			for _, w := range tc.workloads {
+				cache.AddOrUpdateWorkload(w.DeepCopy())
+			}
+
+			snapshot := cache.Snapshot()
+			if diff := cmp.Diff(tc.wantSnapshot(tc.workloads), snapshot, cmpopts.IgnoreUnexported(Cohort{})); diff != "" {
```

Comment:
Most of the things in Snapshot are just copies of the original objects. We probably care more about the accumulated resources in the cohort and we care less about all the fields in ClusterQueue or Workloads to match.

Maybe we can consider using `cmpopts.IgnoreTypes(WorkloadSpec{}, ClusterQueueSpec{})` to reduce the number of lines. WDYT?